torch/distributed/_tensor/dispatch.py

```diff
@@ -68,20 +66,20 @@ class OpSchema(object):
           placements will get implicitly changed and it's error-prone.
     """
 
-    func_schema: FunctionSchema
+    func_schema: torch._C.FunctionSchema
     args_schema: Tuple[object, ...]
     kwargs_schema: Dict[str, object]
+
     is_inplace: bool = False
     is_out_variant: bool = False
 
+
     def __post_init__(self) -> None:
-        schema_kind = self.func_schema.kind()
-        self.is_inplace = (
-            schema_kind == SchemaKind.inplace  # pyre-ignore [16] pyre bad at enum
-        )
-        self.is_out_variant = (
-            schema_kind == SchemaKind.out  # pyre-ignore [16] pyre bad at enum
-        )
+        # simple analysis of function schema to determine
```

Comment:
Conceptual: Does this PR regress the computation of `self.is_inplace` and `self._is_out_variant` in some cases because you are using a heuristic instead of `SchemaKind`?

I am curious how you think about the tradeoff of getting rid of the `torchgen` dependency versus possibly having incorrect values for `is_inplace` and `is_out_variant` in some cases (if at all).

Reply:
Good point, from the current operator set registered, i think it should work fine as all of the op signature naming following the same convention for inplace and out variant (`_` and `.out`), so I think this heuristic is relatively safe to use and should not regress existing registered operators. However, it does rely on operators following the same convention of naming, which I think it might get changed. Given that ATen op set is relatively stable at this point, I think we can go with this approach as it's simple and very fast without parsing FunctionSchema, if ever op signature gets changed, the heuristic need to be adapted to accommodate.